assignments/lambda-classes.js

```diff
@@ -1 +1,139 @@
 // CODE here for your Lambda Classes
+
+class Person {
+    constructor(attributes) {
+        this.name = attributes.name;
+        this.age = attributes.age;
+        this.location = attributes.location;
+        this.gender = attributes.gender;
+    }
+    speak() {
+        console.log(`Hello my name is ${this.name}, I am from ${this.location}.`);
+    }
+}
+
+class Instructor extends Person {
+    constructor(instAttributes) {
+        super(instAttributes);
+        this.specialty = instAttributes.specialty;
+        this.favLanguage = instAttributes.favLanguage;
+        this.catchPhrase = instAttributes.catchPhrase;
+    }
+    demo(subject) {
+        console.log(`Today we are learning about ${subject}.`);
+    };
+    grade(student, subject) {
+        console.log(`${student.name} receives a perfect score on ${subject}.`);
+    };
+    calculateGrade(student) {
+        let rando = Math.floor(Math.random() * Math.floor(10));
+        if (rando %2 === 0) {
+            return student.grade = (student.grade - rando);
+        } else if (rando %2 !== 0) {
+            return student.grade = (student.grade + rando);
+        }
+    };
+}
+
+class Student extends Person {
+    constructor(studAttributes) {
+        super(studAttributes);
+        this.previousBackground = studAttributes.previousBackground;
+        this.className = studAttributes.className;
+        this.favSubjects = studAttributes.favSubjects;
+        this.grade = studAttributes.grade;
+    }
+    listsSubjects() {
+        console.log(this.favSubjects.toString());
+    };
+    PRAssignment(subject) {
+        console.log(`${this.name} has submitted a PR for ${subject}.`);
+    };
+    sprintChallenge(subject) {
+        console.log(`${this.name} has begun sprint challenge on ${subject}.`);
+    };
+    graduate(teacher) {
+        if (this.grade >= 71) {
+            console.log(`${this.name} is ready to graduate!`);
+        } else if (this.grade < 71) {
+            console.log(`${this.name} needs some more flex-time.`);
+            return teacher.calculateGrade(this.name);
+        }
```

Comment:
Remember that `.calculateGrade` is supposed to work with a `Student` object, rather than a string.

You might have meant to pass in `this` (the entire Student object, via implicit binding) as the argument for this method.